File: android app/app/src/main/java/com/esibil/call/ui/CallRecord.kt
package com.esibil.call.ui

/**
 * Immutable representation of a single call history entry shown in
 * [CallHistoryActivity] / [CallHistoryAdapter]. `duration` is null when the
 * call was not completed (missed / rejected).
 */
data class CallRecord(
    val name: String,
    val prisonerId: String,
    val location: String,
    val date: String,
    val time: String,
    val duration: String?, // null/empty when not applicable
    val status: Status
) {
    enum class Status {
        COMPLETED,
        MISSED,
        REJECTED
    }
}

File: esibil_call_flutter/lib/models/call_record.dart
/// Immutable representation of a single call history entry.
class CallRecord {
  const CallRecord({
    required this.name,
    required this.prisonerId,
    required this.location,
    required this.date,
    required this.time,
    required this.duration,
    required this.status,
  });

  final String name;
  final String prisonerId;
  final String location;
  final String date;
  final String time;
  final String? duration;
  final CallStatus status;

  factory CallRecord.fromJson(Map<String, dynamic> json) {
    return CallRecord(
      name: json['name'] as String? ?? 'Unknown',
      prisonerId: json['prisonerId'] as String? ?? '-',
      location: json['location'] as String? ?? '-',
      date: json['date'] as String? ?? '',
      time: json['time'] as String? ?? '',
      duration: json['duration'] as String?,
      status: CallStatus.fromString(json['status'] as String? ?? 'missed'),
    );
  }
}

enum CallStatus {
  completed,
  missed,
  rejected;

  static CallStatus fromString(String value) {
    switch (value.toLowerCase()) {
      case 'completed':
        return CallStatus.completed;
      case 'rejected':
        return CallStatus.rejected;
      default:
        return CallStatus.missed;
    }
  }

  String get label {
    switch (this) {
      case CallStatus.completed:
        return 'Completed';
      case CallStatus.missed:
        return 'Missed';
      case CallStatus.rejected:
        return 'Rejected';
    }
  }
}
